src/data_switch.rs

```diff
@@ -68,6 +73,14 @@ pub struct Timerange {
     pub end: Timestamp,
 }
 
+/// Specifier of which data to fetch from a source by time, and time resolution
+pub struct TimeSpec {
+    /// The range in time of data to fetch
+    pub timerange: Timerange,
+    /// The time resolution of data that should be fetched
+    pub time_resolution: chronoutil::RelativeDuration,
```

Comment:
```suggestion
    pub time_resolution: RelativeDuration,
```
Since we already import it

Reply:
Done